amethyst/src/main/java/com/vitorpamplona/amethyst/ui/actions/NewPostView.kt

```diff
@@ -623,6 +642,64 @@ private fun BottomRowActions(postViewModel: NewPostViewModel) {
     }
 }
 
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun TakePictureButton(onPictureTaken: (Uri) -> Unit) {
+    var imageUri by remember { mutableStateOf<Uri?>(null) }
+    val launcher =
+        rememberLauncherForActivityResult(
+            contract = ActivityResultContracts.TakePicture(),
+        ) { success ->
+            if (success) {
+                imageUri?.let {
+                    onPictureTaken(it)
+                }
+            }
+        }
+    val context = LocalContext.current
+    val cameraPermissionState =
+        rememberPermissionState(
+            Manifest.permission.CAMERA,
+        )
+
+    Box {
+        IconButton(
+            modifier = Modifier.align(Alignment.Center),
+            onClick = {
+                if (cameraPermissionState.status.isGranted) {
+                    val timeStamp: String = SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US).format(Date())
+                    val storageDir: File? = context.getExternalFilesDir(Environment.DIRECTORY_PICTURES)
+                    File
+                        .createTempFile(
```

Comment:
Hmm, I wonder if it's a good idea to do this here, since it will run on the UI thread. Please have you observed any performance issues?

Reply:
I didn't test the performance yet. I had to stop working on this to fix a few bugs in amber and citrine.
I'll start working on this again tomorrow

Reply:
> I didn't test the performance yet. I had to stop working on this to fix a few bugs in amber and citrine. I'll start working on this again tomorrow

Got it. :+1: